easypermissions/src/main/java/pub/devrel/easypermissions/PermissionRequest.java

```diff
@@ -0,0 +1,269 @@
+package pub.devrel.easypermissions;
+
+import android.app.Activity;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.RestrictTo;
+import android.support.annotation.Size;
+import android.support.annotation.StringRes;
+import android.support.annotation.StyleRes;
+import android.support.v4.app.Fragment;
+
+import java.util.Arrays;
+
+import pub.devrel.easypermissions.helper.PermissionHelper;
+
+/**
+ * An immutable model representing a permission request.
```

Comment:
nit: we can probably say something more descriptive here like "A model object that holds all of the parameters associate with a permission request, such as the request code or the rationale"

Reply:
👍